src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1186,6 +1188,20 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions oldOptions, ParseOptions newOptions, Project project)
+        {
+            var parseOptionsService = project.LanguageServices.GetService<IParseOptionsService>();
+            if (parseOptionsService == null)
+            {
+                return false;
+            }
+
+            var newLanguageVersion = parseOptionsService.GetLanguageVersion(newOptions);
+            var updated = parseOptionsService.WithLanguageVersion(oldOptions, newLanguageVersion);
+
+            return newOptions == updated; // is LanguageVersion the only difference between oldOptions and newOptions?
```

Comment:
I don't like comments like this (that just describe the operation).  INstead, just have a comment above 'var newLanguageVersion' tha tsays something like "Currently, only changes to the LanguageVersion of parse options are supported.  So if this is all that changed, we can proceed.  Otherwise, this change is not allowed."